src/libCrypto/EthCrypto.cpp

```diff
@@ -0,0 +1,183 @@
+
+// https://stackoverflow.com/questions/57385412/
+void SetOpensslSignature(const std::string& sSignatureInHex, ECDSA_SIG* pSign) {
+  // Openssl uses raw pointers and macros for freeing but can force it into
+  // smart pointers methodology
+  std::unique_ptr<BIGNUM, std::function<void(BIGNUM*)>> rr(
+      NULL, [](BIGNUM* b) { BN_free(b); });
+  BIGNUM* r_ptr = rr.get();
+  std::unique_ptr<BIGNUM, std::function<void(BIGNUM*)>> ss(
+      NULL, [](BIGNUM* b) { BN_free(b); });
+  BIGNUM* s_ptr = ss.get();
+
+  std::string sSignatureR =
+      sSignatureInHex.substr(0, sSignatureInHex.size() / 2);
+  std::string sSignatureS = sSignatureInHex.substr(sSignatureInHex.size() / 2);
+
+  BN_hex2bn(&r_ptr, sSignatureR.c_str());
+  BN_hex2bn(&s_ptr, sSignatureS.c_str());
+
+  ECDSA_SIG_set0(pSign, r_ptr, s_ptr);
+}
+
+// This function needs to set the key but also needs to decomprss the
+// coordinates using EC_POINT_set_compressed_coordinates_GFp sPubKeyString
+// should be hex \0 terminated string with no compression bit on it
+bool SetOpensslPublicKey(const char* sPubKeyString, EC_KEY* pKey) {
+  // X co-ordinate
+  std::unique_ptr<BIGNUM, std::function<void(BIGNUM*)>> gx(
+      NULL, [](BIGNUM* b) { BN_free(b); });
+
+  BIGNUM* gx_ptr = gx.get();
+
+  EC_KEY_set_asn1_flag(pKey, OPENSSL_EC_NAMED_CURVE);
+
+  // From
+  // https://www.oreilly.com/library/view/mastering-ethereum/9781491971932/ch04.html
+  // The first byte indicates whether the y coordinate is odd or even
+  int y_chooser_bit = 0;
+
+  if (sPubKeyString[1] == '2') {
+    y_chooser_bit = 0;
+  } else if (sPubKeyString[1] == '3') {
+    y_chooser_bit = 1;
+  } else {
+    std::cout << "Received badly set signature bit! Should be 2 or 3 and got: "
+              << sPubKeyString[1] << std::endl;
+  }
+
+  // Don't want the first byte
+  if (!BN_hex2bn(&gx_ptr, sPubKeyString + 2)) {
+    std::cout << "***** Error getting to x binary format" << std::endl;
+  }
+
+  // Create a new curve group
+  // Refactor: probably can have static/constexpr curve, need to check
+  std::unique_ptr<EC_GROUP, std::function<void(EC_GROUP*)>> curve_group(
+      EC_GROUP_new_by_curve_name(NID_secp256k1),
+      [](EC_GROUP* g) { EC_GROUP_free(g); });
+
+  std::unique_ptr<EC_POINT, std::function<void(EC_POINT*)>> point(
+      EC_POINT_new(curve_group.get()), [](EC_POINT* g) { EC_POINT_free(g); });
+
+  // This performs the decompression at the same time as setting the pubKey
+  EC_POINT_set_compressed_coordinates_GFp(curve_group.get(), point.get(),
+                                          gx_ptr, y_chooser_bit, NULL);
+
+  if (!EC_KEY_set_public_key(pKey, point.get())) {
+    std::cout << "****** ERROR! setting public key attributes" << std::endl;
+  }
+
+  if (EC_KEY_check_key(pKey) == 1) {
+    return true;
+  } else {
+    std::cout << "ec key invalid " << std::endl;
+    return false;
+  }
+}
+
+bool VerifyEcdsaSecp256k1(const std::string&, const std::string& sSignature,
```

Comment:
please document the 1st argument (either /* randomNumber */, or use one of the ways to tame the args unused  warnings.

Why is it unused btw, and why do you need it then?

Reply:
Sure! It is unused because it will be the data going into keccak that follows the data (typed data / EIP-712)